java/code/src/com/suse/manager/reactor/messaging/RegisterMinionEventMessageAction.java

```diff
@@ -115,8 +118,11 @@ public RegisterMinionEventMessageAction(SaltService saltService) {
      */
     @Override
     public void execute(EventMessage msg) {
-        registerMinion(((RegisterMinionEventMessage) msg).getMinionId(), false,
-                empty(), empty());
+        RegisterMinionEventMessage registerMinionEventMessage = ((RegisterMinionEventMessage) msg);
+        Optional<MinionStartupGrains> startupGrainsOpt = Opt.or(registerMinionEventMessage.getMinionStartupGrains(),
+                () -> SALT_SERVICE.getGrains(registerMinionEventMessage.getMinionId(),
```

Comment:
That's correct. We save 1 salt call in case of already registered minions, even if startup grains are not configured.

So if start-up grains are **supported**, with this PR number of salt sync calls are following


Test                                                |Before |After 
|-----------------------------------------------------|----------|----------|
| New registration                     | 2 | 1 |
| Registered minion startup                        | 2 | 0 |

If start-up grains are **not supported**, with this PR number of salt sync calls are following

Test                                                |Before |After 
|-----------------------------------------------------|----------|----------|
| New registration                     | 2 | 2 |
| Registered minion startup                        | 2 |  1 |

Reply:
Please add this relevant information in the PR description, so it does not get lost!